### PR TITLE
Post GitHub comments on changed lines only

### LINKED ISSUE #203: Filter comments outside of the exact lines with changes
Related: #185.

### PR DESCRIPTION
**Pull request by carlosms**
Fix #203.

### CODE REVIEW

**Comment by carlosms** [comment]
Lookout analysis failed with:
```
app: lookout
error: 'posting analysis failed: file not found: .helm-staging.yml'
event-id: f2a5a2f8cad1a28637476806c319bd2ed4bc74e6
event-type: 2
level: error
msg: 'event processing failed'
source: 'server/server.go:101'
time: '2018-08-30T14:27:15.663553209Z'
```

I'm pretty sure the problem is #197, this branch is not a fast-forward of master.
```
$ git diff --stat HEAD^
 provider/github/diff.go      | 68 ++++++++++++++++++++++++++++++++++++++++-------------------
 provider/github/diff_test.go | 84 +++++++++++++++++++++++++++++++++++++++++++++++--------------------------
 provider/github/poster.go    | 10 +++++++--
 3 files changed, 109 insertions(+), 53 deletions(-)

$ git diff --stat origin/master 
 .helm-staging.yml            |  2 --
 .lookout.yml                 |  3 ---
 provider/github/diff.go      | 68 ++++++++++++++++++++++++++++++++++++++++-------------------
 provider/github/diff_test.go | 84 +++++++++++++++++++++++++++++++++++++++++++++++--------------------------
 provider/github/poster.go    | 10 +++++++--
 5 files changed, 109 insertions(+), 58 deletions(-)

```

**Comment by bzz** [comment]
Strict mode filters out everything, that is not on the `+ line` - that indeed sounds strict!

Is it because all current analyzers work with Head, that we did not have issue posting on `- lines` yet?

Otherwise I guess a simpler fix would be possible, by just filtering out the comments that are on context lines.

**Comment by bzz** [comment]
BTW, @carlosms I think CI is failing on code generation profile

```
0.04s$ make no-changes-in-commit
diff --git a/Gopkg.lock b/Gopkg.lock
index 66700ef..57c7856 100644
--- a/Gopkg.lock
+++ b/Gopkg.lock
@@ -665,6 +665,7 @@
     "github.com/jessevdk/go-flags",
     "github.com/lib/pq",
     "github.com/satori/go.uuid",
+    "github.com/stretchr/testify/assert",
     "github.com/stretchr/testify/mock",
     "github.com/stretchr/testify/require",
     "github.com/stretchr/testify/suite",
 M Gopkg.lock
generated assets are out of sync
make: *** [no-changes-in-commit] Error 2
The command "make no-changes-in-commit" exited with 2.
```

File: provider/github/diff.go [hunk 1]
```diff
@@ -45,20 +48,41 @@ type posRange struct {
...
+
+	diffLine, err := d.convertLine(parsedFile.ranges, line)
 	if err != nil {
 		return 0, err
 	}
 
-	return d.convertLine(ranges, line)
+	if strict {
+		if !parsedFile.linesAdded[diffLine] {
+			return 0, ErrLineNotAddition.New()
```
Comment (bzz) [inline provider/github/diff.go]: Is there a chance we could pass actual file name and line number to the error? 
I find having such information invaluable when debugging something later

Comment (carlosms) [reply provider/github/diff.go]: We log that debug info, but in poster.go: https://github.com/src-d/lookout/pull/210/files#diff-874f7f13213445f79bf1b3482a66e541R181

**Comment by carlosms** [comment]
Integration tests are failing because of https://github.com/src-d/lookout/issues/215.
Since tests pass locally, and we should not fix that in this PR, I'll go ahead and merge.

**Pull request merged**